backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Boxing.kt

```diff
@@ -42,8 +42,7 @@ private fun KonanSymbols.getTypeConversionImpl(
 internal object DECLARATION_ORIGIN_INLINE_CLASS_SPECIAL_FUNCTION : IrDeclarationOriginImpl("INLINE_CLASS_SPECIAL_FUNCTION")
 
 internal val Context.getBoxFunction: (IrClass) -> IrSimpleFunction by Context.lazyMapMember { inlinedClass ->
-    assert(inlinedClass.isUsedAsBoxClass())
-    assert(inlinedClass.parent is IrFile) { "Expected top level inline class" }
+    require(inlinedClass.isUsedAsBoxClass())
```

Comment:
Btw, we still rely on inline classes being non-local (more specifically, we expect that inline class is nested to the chain of classes, last of which is top-level).